src/extension/utils/confFileToInputForm.ts

```diff
@@ -0,0 +1,158 @@
+/* ---------------------------------------------------------------------------------------------
+ *  Convert conf file to input form
+ *-------------------------------------------------------------------------------------------- */
+
+import { nanoid } from 'nanoid'
+import type { InputFormData, ConfFile } from '../types'
+
+const emptyForm: InputFormData = {
+  mainSolidityFile: '',
+  mainContractName: '',
+  specFile: '',
+  solidityCompiler: '',
+  useAdditionalContracts: false,
+  additionalContracts: [],
+  link: [],
+  extendedSettings: [{ id: nanoid(), flag: '' }],
+  useStaging: false,
+  branch: 'master',
+  cacheName: '',
+  message: '',
+  additionalSettings: [
+    {
+      id: nanoid(),
+      option: '',
+      value: '',
+    },
+  ],
+  name: '',
+  solc_map: [],
+}
+
+const stableFields = [
+  'files',
+  'verify',
+  'solc',
+  'link',
+  'settings',
+  'staging',
+  'cache',
+  'msg',
+]
+
+function getAdditionalSettings(confFile: ConfFile) {
+  const copy = { ...confFile }
+
+  stableFields.forEach(field => {
+    delete copy[field]
+  })
+
+  return copy
+}
+
+/**
+ * converts conf file to input form
+ * @param confFile to convert
+ * @param name of the run
+ * @returns input form
+ */
+export function confFileToFormData(
+  confFile: ConfFile,
+  name: string,
+): InputFormData {
+  const form = emptyForm as InputFormData
+
+  form.name = name
+
+  // convert solidity file / contract
+  if (Array.isArray(confFile.files) && confFile.files.length > 0) {
+    form.mainSolidityFile = confFile.files[0] as string
+
+    if (form.mainSolidityFile.includes(':')) {
+      form.mainSolidityFile = form.mainSolidityFile.split(':')[0]
+    }
+
+    if (confFile.files.length > 1) {
+      const [, ...additional] = confFile.files
+
+      form.useAdditionalContracts = true
+      form.additionalContracts = additional.map(contract => {
+        const [file, name] = contract.split(':')
+
+        if (file && name) {
+          return {
+            file,
+            name,
+          }
+        }
+
+        return {
+          file: contract,
+          name: '',
+        }
+      })
+    }
+  }
+
+  if (Array.isArray(confFile.verify) && confFile.verify.length === 1) {
+    const verifyStr = confFile.verify[0] as string
+    const [mainContractName, specFile] = verifyStr.split(':')
+
+    if (mainContractName) {
+      form.mainContractName = mainContractName
+    }
+    if (specFile) {
+      form.specFile = specFile
+    }
+  }
+
+  if (confFile.solc) {
+    form.solidityCompiler = confFile.solc as string
+  }
+
+  if (Array.isArray(confFile.link) && confFile.link.length > 0) {
+    confFile.link.forEach(link => {
+      const linkArr = link.split(/[:=]/)
```

Comment:
- add comment about looking for a specific format
- add to constants (format of link) - later

Reply:
* added comment to link format
* added constant file assingment to jira